packages/opentelemetry-api/src/diag/logger.ts

```diff
@@ -0,0 +1,158 @@
+  error: DiagLogFunction;
+
+  /** Log a warning scenario to inform the developer of an issues that should be investigated.
+   * The requested operation may or may not have succeeded or completed.
+   */
+  warn: DiagLogFunction;
+
+  /** Log a general informational message, this should not affect functionality.
+   * This is also the default logging level so this should NOT be used for logging
+   * debugging level information.
+   */
+  info: DiagLogFunction;
+
+  /** Log a general debug message that can be useful for identifying a failure.
+   * Information logged at this level may include diagnostic details that would
+   * help identify a failure scenario. Useful scenarios would be to log the execution
+   * order of async operations
+   */
+  debug: DiagLogFunction;
+
+  /** Log a detailed (verbose) trace level logging that can be used to identify failures
+   * where debug level logging would be insufficient, this level of tracing can include
+   * input and output parameters and as such may include PII information passing through
+   * the API. As such it is recommended that this level of tracing should not be enabled
+   * in a production environment.
+   */
+  trace: DiagLogFunction;
+
+  /** Log a general informational message that should always be logged regardless of the
+   * current {@Link DiagLogLevel) and configured filtering level. This type of logging is
+   * useful for logging component startup and version information without causing additional
+   * general informational messages when the logging level is set to DiagLogLevel.WARN or lower.
+   */
+  forcedInfo: DiagLogFunction;
+}
+
+const loggerMapFuncs: { d: keyof DiagLogger; l: keyof Logger }[] = [
+  { d: 'trace', l: 'debug' },
+  { d: 'debug', l: 'debug' },
+  { d: 'info', l: 'info' },
+  { d: 'warn', l: 'warn' },
+  { d: 'error', l: 'error' },
+  { d: 'critical', l: 'error' },
+  { d: 'terminal', l: 'error' },
+  { d: 'forcedInfo', l: 'info' },
+];
+
+function noopLogFunction() {}
+
+/**
+ * Returns a No-Op Diagnostic logger where all messages do nothing.
+ * @implements {@link DiagLogger}
+ * @returns {DiagLogger}
+ */
+export function noopDiagLogger(): DiagLogger {
+  const diagLogger = {} as DiagLogger;
+
+  for (let lp = 0; lp < loggerMapFuncs.length; lp++) {
+    const map = loggerMapFuncs[lp];
+    diagLogger[map.d] = noopLogFunction;
+  }
+
+  return diagLogger;
+}
+
+/**
+ * @deprecated This helper will return either the original passed logger if it implements {@link DiagLogger} a
+ * wrapped instance if it only implements {@link Logger} or null if it's null, undefined.
+ * This helper will be removed as part of the {@link Logger} deprecation, please use {@link DiagLogger} from the api.
+ * @see {@link DiagLogLevel} and {@link DiagLogger}
+ * @returns The passed Logger wrapped as a DiagLogger or null if no Logger was supplied.
+ */
+export function diagLoggerAdapter(
```

Comment:
This helper ONLY exists to help speed up migration so we can get to Part 2 faster (hopefully).

Reply:
Removed